### PR TITLE
Create edit person page

### PR DESCRIPTION
**Pull request by cdra334**
* Created create/edit person page
* Automatically changes header depending on route
* Allow the change of profile picture
* Allow additions of social media links through a modal

TODO: 
* implement custom input field component when finished(#63)
* properly connect to API calls. Mostly setup already
* Create more tests, especially for API calls
* Properly link save to next page (create encounter page)

Create Person page:
image
Edit Person page:
image
Social media modal:
image

Refers to #50

### CODE REVIEW

**Review by uoa-lchu090 — APPROVED** [review]
LGTM, great job on this!

File: forgettable-frontend/src/pages/edit/EditPerson.js [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+      image: 'iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAIAAAD/gAIDAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAI2SURBVHhe7dpBcgFBFIdxspqlJTtu4hhuwC0cgx23MMexdAPLyavqri6FZOYfyWuv8v0WySBJma9aT6cZd103wjAf+TsGIJaAWAJiCYglIJaAWAJiCYglIJaAWAJiCYglIJaAWAJiCYglIJaAWAJiCYglIJaAWAJiCf5LrLZtF4vF+IHdeTwe8w/16iI7nU7L5fJwOOTbN+yh+XyeT/JbTdPk3+kTO9Z0OrWztQGSTvtnNptN/nN9gsUaPl6eGt7lqQCxXgn0Yp077xtLavS7Ub7yprGs1GQyySUelDT5ttdl6k1jPY6pp2MnP+YV603XWev1Oh2URrvdLt1TUexP/pVFg89Z8O+OgFgCYgmIJQgcq23bfOQl8NVwNptdLpd0zNWwRylla7F08NcCjyznRZZhghdEjeU/u5uoL0P/2d1EHVn+s7uJOrL8Z3fDBC8IGWu/3+cjZzaMw0nvgJnVapXvchFyzioT1vV6bZomHTuIHcv5ycebszyXo/efkAg3sjyXo1bqfD7nGxFHludytLzJlMQbWfZqSAf+z5xFqSBYrCqbDUWwl2GVzYYi2MiqstlQBBtZFWd3wwQvIJaAWAJiCSLFqrbnV9hlJYpae35FpKVDWTc47/kVIWPVes5M8AJiCYgliBSryqR+K1Ks7XZrX6vsNyTxtpUrYs4SEEtALAGxBMQSEEtALAGxBMQSEEtALAGxBMQSEEtALAGxBMQSEEtALAGxBMQSEEtALAGxBhuNPgHlJKV46HCjogAAAABJRU5ErkJggg==',
+      encounters: [],
+      time_updated: '0002-02-02',
+    });
+
+  async function getPersonData() {
+    try {
+      personData = await apiCalls.getPerson(id);
+    } catch (err) {
+      // console.log(err);
```
Comment (ckar364) [inline forgettable-frontend/src/pages/edit/EditPerson.js:47]: Why is line 47 commented out, after we catch the error?

Comment (cdra334) [reply forgettable-frontend/src/pages/edit/EditPerson.js:47]: The API calls aren't properly implemented yet. Having a console log probably isn't too helpful either. Perhaps later, a proper message to the user is displayed. I guess for MVP a console log might be okay.

Comment (ckar364) [reply forgettable-frontend/src/pages/edit/EditPerson.js:47]: For consistency with the other functions in this file, let's have the error log to the console for now.

**Review by ckar364 — CHANGES_REQUESTED** [review]
Good job, the pages look great. I would suggest changing line 47 of the EditPerson.js file to log the error to the console, as this makes it consistent with other functions in the file. I liked seeing the use of a constant to represent MAX_IMAGE_SIZE to avoid magic numbers.

**Comment by jyao244** [comment]
well done
LGTM

**Review by jyao244 — DISMISSED** [review]
well done!
LGTM

**Review by ckar364 — DISMISSED** [review]
LGTM

**Review by noahjade — DISMISSED** [review]
This looks great!! Thanks for the hard work! 💪 
Small non blocking stuff but just noting so we don't forget about it:
- I noticed that your routes are people/ but jiaru's person view one is person/:id. Thats just a style thing tho so non blocking, but maybe we could make a tech debt ticket to unite terminology in the future to make things clean for an end user.
- Similar, after the user clicks submit are we navigating them away from the page? What happens when someone deletes a person? Do they get navigated away? Sorry if I'm just not seeing that logic haha

**Comment by cdra334** [comment]
> This looks great!! Thanks for the hard work! 💪 Small non blocking stuff but just noting so we don't forget about it:
> 
> * I noticed that your routes are people/ but jiaru's person view one is person/:id. Thats just a style thing tho so non blocking, but maybe we could make a tech debt ticket to unite terminology in the future to make things clean for an end user.
> * Similar, after the user clicks submit are we navigating them away from the page? What happens when someone deletes a person? Do they get navigated away? Sorry if I'm just not seeing that logic haha

* Good point about the route inconsistencies. I guess perhaps perhaps do that later on? unless you think it is very important.
* I realised that the delete person button should also have a confirmation modal, so I will work on that too. Yes sorry, I think I forgot to navigate it to anywhere.

**Review by ckar364 — APPROVED** [review]
LGTM

File: forgettable-frontend/src/pages/edit/EditPerson.js [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+    organisation: 'job co.',
+    social_media: new Map([['twitter', 'fdfd']]),
+    image: 'iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAIAAAD/gAIDAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAI2SURBVHhe7dpBcgFBFIdxspqlJTtu4hhuwC0cgx23MMexdAPLyavqri6FZOYfyWuv8v0WySBJma9aT6cZd103wjAf+TsGIJaAWAJiCYglIJaAWAJiCYglIJaAWAJiCYglIJaAWAJiCYglIJaAWAJiCYglIJaAWAJiCf5LrLZtF4vF+IHdeTwe8w/16iI7nU7L5fJwOOTbN+yh+XyeT/JbTdPk3+kTO9Z0OrWztQGSTvtnNptN/nN9gsUaPl6eGt7lqQCxXgn0Yp077xtLavS7Ub7yprGs1GQyySUelDT5ttdl6k1jPY6pp2MnP+YV603XWev1Oh2URrvdLt1TUexP/pVFg89Z8O+OgFgCYgmIJQgcq23bfOQl8NVwNptdLpd0zNWwRylla7F08NcCjyznRZZhghdEjeU/u5uoL0P/2d1EHVn+s7uJOrL8Z3fDBC8IGWu/3+cjZzaMw0nvgJnVapXvchFyzioT1vV6bZomHTuIHcv5ycebszyXo/efkAg3sjyXo1bqfD7nGxFHludytLzJlMQbWfZqSAf+z5xFqSBYrCqbDUWwl2GVzYYi2MiqstlQBBtZFWd3wwQvIJaAWAJiCSLFqrbnV9hlJYpae35FpKVDWTc47/kVIWPVes5M8AJiCYgliBSryqR+K1Ks7XZrX6vsNyTxtpUrYs4SEEtALAGxBMQSEEtALAGxBMQSEEtALAGxBMQSEEtALAGxBMQSEEtALAGxBMQSEEtALAGxBhuNPgHlJKV46HCjogAAAABJRU5ErkJggg==',
+    encounters: [],
+    time_updated: '0002-02-02',
+  });
+
+  // const [personData, setPersonData] = useState({
+  //   first_name: '',
+  //   last_name: '',
```
Comment (noahjade) [inline forgettable-frontend/src/pages/edit/EditPerson.js:43]: Is there a reason you stopped using the blank useState?

**Review by noahjade — DISMISSED** [review]
Looks amazing!! So much good work here 🎉 🎉

File: forgettable-frontend/src/pages/edit/EditPerson.js (hunk 2 above)
Comment (cdra334) [reply forgettable-frontend/src/pages/edit/EditPerson.js:43]: no real reason. I can change it back if you want me to.

**Review by ckar364 — APPROVED** [review]
LGTM

**Review by noahjade — APPROVED** [review]
Such awesome work on this!!! Thank you!!!

**Pull request merged**